### DISCUSSION TITLE
Why is Moonraker transmitting almost 750mb of data during update?

### DISCUSSION THREAD

**Discussion by HcKaos** [Arksine/moonraker#891] [Q&A]
So i need an explenaition why, what and where moonraker is transmitting almost 800mb of data during the update? 
![moonraker](https://github.com/user-attachments/assets/ca35641c-a4b9-4926-ba5f-6f32fab87b82)
This cant be normal.

**Comment by Arksine** [comment]
I have no idea what that image is you posted or where its getting its information, but I suspect it is in error.  The only data Moonraker would transmit during an update would be the current status to the frontend.

Edit: I suspect that data is the transmission data reported from the system.  That is total data transmitted by the SBC/PC since its last boot, not the amount transmitted by Moonraker during an update.

**Comment by HcKaos** [comment]
it took 15 min to update for about 60mb update file. while the update it started to transmit 741.5mb data somewhere and after the upload was done the update finished and transmittion stopped. 
BTW: the image is out of moonraker update

**Reply by Arksine** [reply]
I'm not sure what "moonraker update" is, but its not a part of the software in this repo.  I presumed that the image came from Mainsail or Fluidd, but that does not seem to be the case based on your response.  Moonraker does have an update manager, capable of self-updating and updating other software such as Klipper, Spoolman, etc. 

Moonraker itself does not transmit data anywhere other than the connected front-ends.  This is the data that front-ends use to display state.  This would not amount to over 700MiB in a short period of time.  In addition, Moonraker does not track or report the amount of data it transmits.  Moonraker can report the amount of data transmitted per network interface.  In this case, the amount reported is the total data transmitted/received on the interface since the last boot.  

If you are running a fork or unofficial extensions then perhaps  they are transmitting.  A 60MiB total download during an update may be possible, depending on how many application are configured for update.  15 minutes to update may be possible if a dependency needs to be built.  I'm not aware of any way a 741MiB transmission is possible by the code in this repo.  Moonraker is 100% free and open source, so there is no need to take my word for it, anyone can inspect the code.

**Comment by Ba22a** [comment]
Could it be Obico? That is called update manager in the update section. That obviously uploads for monitoring purposes.


Regards
Barry

On Saturday, July 27, 2024, 9:58 PM, Eric Callahan ***@***.***> wrote:




I'm not sure what "moonraker update" is, but its not a part of the software in this repo. I presumed that the image came from Mainsail or Fluidd, but that does not seem to be the case based on your response. Moonraker does have an update manager, capable of self-updating and updating other software such as Klipper, Spoolman, etc.

Moonraker itself does not transmit data anywhere other than the connected front-ends. This is the data that front-ends use to display state. This would not amount to over 700MiB in a short period of time. In addition, Moonraker does not track or report the amount of data it transmits. Moonraker can report the amount of data transmitted per network interface. In this case, the amount reported is the total data transmitted/received on the interface since the last boot.

If you are running a fork or unofficial extensions then perhaps they are transmitting. A 60MiB total download during an update may be possible, depending on how many application are configured for update. 15 minutes to update may be possible if a dependency needs to be built. I'm not aware of any way a 741MiB transmission is possible by the code in this repo. Moonraker is 100% free and open source, so there is no need to take my word for it, anyone can inspect the code.

**Comment by HcKaos** [comment]
Some more details.
i got  a creality k1 printer and i use the https://guilouz.github.io/Creality-Helper-Script-Wiki/ with mainsail. in mainsail/machine tab i got a update Manager where i can update the helper script, mainsail, moonraker and system. 
![mainsail](https://github.com/user-attachments/assets/522a12bf-73ab-432f-a89b-df68f1a389c9)

i am not sure if it is just a bug or if something fishy is going on (all the linux hacks recently) so i am a bit worried. Not saying it has something to do with moonraker but the big transmittion happend on the update of moonraker. i know if i for example have a window with the webcam feed open it will amount to a lot of transmitted data in a short time between host an client, but at this time i had only the update window open so it should not be the webcam feed.
